jsonrpc/debug_endpoint.go

```diff
@@ -273,6 +276,32 @@ func (d *Debug) GetRawHeader(filter BlockNumberOrHash) (interface{}, error) {
 	)
 }
 
+// GetRawTransaction returns the bytes of the transaction for the given hash.
+func (d *Debug) GetRawTransaction(txHash types.Hash) (interface{}, error) {
+	return d.throttling.AttemptRequest(
+		context.Background(),
+		func() (interface{}, error) {
+			tx, block := GetTxAndBlockByTxHash(txHash, d.store)
+
+			if tx == nil {
+				if txPending, ok := d.store.GetPendingTx(txHash); ok {
+					tx = txPending
+				}
+			}
+
+			if tx == nil {
+				return nil, fmt.Errorf("tx %s not found", txHash.String())
```

Comment:
Should we follow the Geth implementation here? They are not returning any errors (https://github.com/ethereum/go-ethereum/blob/9b5d1412cce142bad88455d71fcb82cc91efb946/internal/ethapi/api.go#L2093-L2095). I mean we can do whatever we agree I suppose.

```suggestion
				return nil, nil
```

Reply:
I think we should be consistent with eth_endpoint GetTransactionByHash implementation. That method works as Geth, i.e. it doesn't return error if tx is not found. So I agree with Stefan to follow Geth, actually to follow our own GetTransactionByHash implementation.